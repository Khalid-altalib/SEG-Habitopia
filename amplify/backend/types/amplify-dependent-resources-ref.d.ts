export type AmplifyDependentResourcesAttributes = {
  "api": {
    "Habitopia": {
      "GraphQLAPIEndpointOutput": "string",
      "GraphQLAPIIdOutput": "string",
      "GraphQLAPIKeyOutput": "string"
    }
  },
  "auth": {
    "Habitopia": {
      "AppClientID": "string",
      "AppClientIDWeb": "string",
      "IdentityPoolId": "string",
      "IdentityPoolName": "string",
      "UserPoolArn": "string",
      "UserPoolId": "string",
      "UserPoolName": "string"
    }
  }
}