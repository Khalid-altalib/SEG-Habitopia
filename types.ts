import type { NativeStackNavigationProp } from "@react-navigation/native-stack";


export type RootParams = {
  Auth: undefined,
  Existing: undefined,
  Chat : undefined,
}

export type ProfileParams = {
  Home: undefined;
  Chats: undefined;
  Welcome: undefined;
  Explore: undefined;
  Leaderboard: undefined;
  You: undefined;
  Chat: undefined;
}

export type AuthParams = {
  Welcome: undefined;
  SignUp: undefined;
  LogIn: undefined;
};

export type ChatParams= {
  Chat: undefined;
}


export type LocalUser = {
  authToken: string;
  username: string;
};

