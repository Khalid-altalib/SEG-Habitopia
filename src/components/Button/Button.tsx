import React from "react";
import { StyleSheet, TouchableOpacity, View, ViewStyle } from "react-native";

import Text from "../Text";
import { ButtonType, TextType } from "../../../types";
import { theme, useColorModeValue } from "native-base";
import { LinearGradient } from "expo-linear-gradient";

type Props = {
  style?: ViewStyle;
  isFullWidth?: boolean;
  onPress?: () => void;
  type: ButtonType;
  icon?: JSX.Element;
  children: string;
};

const Button = ({
  style,
  isFullWidth = false,
  onPress,
  type,
  icon,
  children,
}: Props) => {
  if (type == undefined) type = ButtonType.Primary;

  const styles = StyleSheet.create({
    base: {
      borderRadius: 10,
      paddingVertical: 15,
      paddingHorizontal: 40,
      alignItems: "center",
      justifyContent: "center",
    },
  });

  const typeStyles = {
    primary: {
      backgroundColors: [theme.colors.darkBlue[500], theme.colors.purple[700]],
    },
    secondary: {
      backgroundColors: useColorModeValue(
        [theme.colors.gray[500], theme.colors.gray[400]],
        [theme.colors.blueGray[600], theme.colors.blueGray[700]]
      ),
    },
  };

  const additionalStyle = ButtonType[type]
    .toString()
    .toLowerCase() as keyof typeof typeStyles;

  return (
    <View
      style={[
        {
          shadowColor: "black",
          shadowOffset: { width: 5, height: 5 },
          shadowOpacity: 0.15,
          borderRadius: 10,
        },
      ]}
    >
      <TouchableOpacity activeOpacity={0.7} onPress={onPress} style={style}>
        <LinearGradient
          colors={typeStyles[additionalStyle].backgroundColors}
          start={[0, 0]}
          end={[1, 1]}
          style={[
            styles.base,
            isFullWidth ? { width: "100%" } : { alignSelf: "flex-start" },
          ]}
        >
          <View style={{ flexDirection: "row", alignItems: "center" }}>
            <Text
              type={TextType.Button}
              style={icon == undefined ? null : { marginRight: 5 }}
            >
              {children}
            </Text>
            {icon}
          </View>
        </LinearGradient>
      </TouchableOpacity>
    </View>
  );
};

export default Button;
