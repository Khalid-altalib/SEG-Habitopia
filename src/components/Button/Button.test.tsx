// React
import React from "react";

// React Native Testing Library
import { render, screen } from "@testing-library/react-native";

// Habitopia
import Button from "@components/Button";
import { ButtonType } from "types";

it("Renders correctly.", () => {
  render(
    <Button type={ButtonType.Primary} isLoading={false}>
      This is some example content.
    </Button>
  );

  expect(screen.toJSON()).toMatchSnapshot();
});
