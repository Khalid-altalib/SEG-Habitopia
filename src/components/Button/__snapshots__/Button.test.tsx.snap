// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly. 1`] = `
<View
  isLoading={false}
  style={
    [
      {
        "borderRadius": 10,
        "shadowColor": "black",
        "shadowOffset": {
          "height": 5,
          "width": 5,
        },
        "shadowOpacity": 0.15,
      },
    ]
  }
  type={0}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4278220774,
            4286456526,
          ],
          "endPoint": [
            1,
            1,
          ],
          "locations": undefined,
          "startPoint": [
            0,
            0,
          ],
        }
      }
      style={
        [
          {
            "alignItems": "center",
            "borderRadius": 10,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          {
            "alignSelf": "flex-start",
          },
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#18181b",
            },
            {
              "color": "#e4e4e7",
              "fontFamily": undefined,
              "fontSize": 19,
            },
            [
              null,
              {
                "alignSelf": "center",
              },
            ],
          ]
        }
      >
        This is some example content.
      </Text>
      <View
        style={
          {
            "marginLeft": 5,
          }
        }
      />
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
</View>
`;
